mono/metadata/sgen-mono.c

```diff
@@ -1252,16 +1252,16 @@ create_allocator (int atype, gboolean slowpath)
 		g_assert_not_reached ();
 	}
 
-	if (atype != ATYPE_SMALL) {
-		/* size += ALLOC_ALIGN - 1; */
 		mono_mb_emit_ldloc (mb, size_var);
-		mono_mb_emit_icon (mb, SGEN_ALLOC_ALIGN - 1);
-		mono_mb_emit_byte (mb, CEE_ADD);
-		/* size &= ~(ALLOC_ALIGN - 1); */
-		mono_mb_emit_icon (mb, ~(SGEN_ALLOC_ALIGN - 1));
-		mono_mb_emit_byte (mb, CEE_AND);
-		mono_mb_emit_stloc (mb, size_var);
 	}
+	/* size += ALLOC_ALIGN - 1; */
```

Comment:
http://xamarin.github.io/benchmarker/front-end/pullrequest.html#id=176
FWIW, another PR with just the removal of zeroing on strings change:
http://xamarin.github.io/benchmarker/front-end/pullrequest.html#id=179

I can't really make heads or tails with these, it seems that neither matters enough to drown the noise. Maybe I should run the bench a couple more times.